logs/ingress/src/audit_log.rs

```diff
@@ -0,0 +1,252 @@
+use backoff;
+use backoff::backoff::Backoff;
+use logs_lib::id;
+use logs_lib::{time, AuditLogEntryType};
+use reqwest::StatusCode;
+use serenity;
+use serenity::http::error::ErrorResponse;
+use serenity::http::Http;
+use serenity::model::guild::AuditLogEntry;
+use serenity::model::id::{AuditLogEntryId, GuildId, UserId};
+use std::sync::Arc;
```

Comment:
If you're using this for async stuff, Tokio implements it's own async safe reference types that are probably better to use.